### PR TITLE
fixed mempool to be generated using gene and allow it to fetch items c…

### PR DESCRIPTION
**Pull request by antonlerner**
…oncurrently

### CODE REVIEW

File: miner/pool.go [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
+		txList = append(txList, *k)
 	}
-	return txList.Interface()
+	return txList
 }
 
-func (mem *MemPool) Put(id interface{}, item interface{}) {
+func (mem *KeyTypeMemPool) Put(id KeyType, item *ValueType) {
 	mem.mu.Lock()
 	defer mem.mu.Unlock()
```
Comment (y0sher) [inline miner/pool.go]: deferring here is redundant since there's only one return path

File: sync/handler.go [hunk 2]
```diff
@@ -83,8 +83,8 @@ func newTxsRequestHandler(s *Syncer, logger log.Log) func(msg []byte) []byte {
 		txs, missinDB := s.GetTransactions(txids)
 
 		for _, t := range missinDB {
-			if tx := s.txpool.Get(t); tx != nil {
-				txs[t] = tx.(*types.AddressableSignedTransaction)
+			if tx, err := s.txpool.Get(t); err == nil {
```
Comment (y0sher) [inline sync/handler.go:86]: why not `tx, err := txpool.Get(); if err != nil { return err }`

File: sync/syncer.go [hunk 3]
```diff
@@ -416,9 +423,9 @@ func (s *Syncer) syncTxs(txids []types.TransactionId) ([]*types.AddressableSigne
 	unprocessedTxs := make(map[types.TransactionId]*types.AddressableSignedTransaction)
 	missing := make([]types.TransactionId, 0)
 	for _, t := range txids {
-		if tx := s.txpool.Get(t); tx != nil {
+		if tx, err := s.txpool.Get(t); err == nil {
 			s.Info("found tx, %v in tx pool", hex.EncodeToString(t[:]))
-			unprocessedTxs[t] = tx.(*types.AddressableSignedTransaction)
+			unprocessedTxs[t] = &tx
```
Comment (y0sher) [inline sync/syncer.go:428]: why do you need the address of tx ? (i know `unprocessedTxs` is id to pointer but maybe you dont need it).

File: sync/handler.go (hunk 2 above)
Comment (antonlerner) [reply sync/handler.go:86]: handlers do not return errors, I wanted to use the if one liner

File: sync/syncer.go (hunk 3 above)
Comment (antonlerner) [reply sync/syncer.go:428]: to keep consistency and not change or refactor any outside dependencies I've decided to keep it this way.

**Review by y0sher — APPROVED** [review]
we also need to make sure these files are generated again when changing. or at least add documentation about generating them.

**Pull request merged**